bundler/lib/bundler/compact_index_client/updater.rb

```diff
@@ -98,6 +97,20 @@ def checksum_for_file(path)
           SharedHelpers.digest(:MD5).hexdigest(File.read(path))
         end
       end
+
+      private
+
+      def copy_file(source, dest)
+        SharedHelpers.filesystem_access(source, :read) do
+          File.open(source, "r") do |s|
+            SharedHelpers.filesystem_access(dest, :write) do
+              File.open(dest, "wb", s.stat.mode) do |f|
+                IO.copy_stream(s, f)
```

Comment:
Just in case someone runs this test on a slightly outdated kernel. There was a bug in Linux 5.10 that caused `IO.copy_stream` to copy 0 bytes. It has since been fixed: https://github.com/docker/for-linux/issues/1015#issuecomment-985799437

Reply:
Ah, of course. So no change in behavior. Great, thanks!